clouddrift/signal.py

```diff
@@ -89,119 +119,187 @@ def analytic_transform(
     else:
         raise ValueError("boundary must be one of 'mirror', 'align', or 'zeros'.")
 
-    if np.isrealobj(xa):
-        # fft should be taken along last axis
-        z = 2 * np.fft.fft(xa)
-    else:
-        z = np.fft.fft(xa)
+    # analytic signal
+    xap = np.fft.fft(xa)
+    # anti-analytic signal
+    xan = np.fft.fft(np.conj(xa))
 
     # time dimension of extended time series
     M = np.shape(xa)[-1]
 
     # zero negative frequencies
     if M % 2 == 0:
-        z[..., int(M / 2 + 2) - 1 : int(M + 1) + 1] = 0
+        xap[..., int(M / 2 + 2) - 1 : int(M + 1) + 1] = 0
+        xan[..., int(M / 2 + 2) - 1 : int(M + 1) + 1] = 0
         # divide Nyquist component by 2 in even case
-        z[..., int(M / 2 + 1) - 1] = z[..., int(M / 2 + 1) - 1] / 2
+        xap[..., int(M / 2 + 1) - 1] = xap[..., int(M / 2 + 1) - 1] / 2
+        xan[..., int(M / 2 + 1) - 1] = xan[..., int(M / 2 + 1) - 1] / 2
     else:
-        z[..., int((M + 3) / 2) - 1 : int(M + 1) + 1] = 0
+        xap[..., int((M + 3) / 2) - 1 : int(M + 1) + 1] = 0
+        xan[..., int((M + 3) / 2) - 1 : int(M + 1) + 1] = 0
 
     # inverse Fourier transform along last axis
-    z = np.fft.ifft(z)
+    xap = np.fft.ifft(xap)
+    xan = np.fft.ifft(xan)
 
-    # return central part
-    z = z[..., int(N + 1) - 1 : int(2 * N + 1) - 1]
+    # return central part plus hlaf the mean
+    xap = xap[..., int(N + 1) - 1 : int(2 * N + 1) - 1] + 0.5 * mx_
+    xan = xan[..., int(N + 1) - 1 : int(2 * N + 1) - 1] + 0.5 * np.conj(mx_)
+
+    if np.isrealobj(x):
+        xa = xap + xan
+    else:
+        xa = (xap, xan)
 
     # return after reorganizing the axes
     if time_axis != -1 and time_axis != len(x.shape) - 1:
-        return np.swapaxes(z, time_axis, -1)
+        return np.swapaxes(xa, time_axis, -1)
     else:
-        return z
+        return xa
 
 
-def rotary_transform(
-    u: Union[np.ndarray, xr.DataArray],
-    v: Union[np.ndarray, xr.DataArray],
-    boundary: Optional[str] = "mirror",
+def cartesian_to_rotary(
+    ua: Union[np.ndarray, xr.DataArray],
+    va: Union[np.ndarray, xr.DataArray],
     time_axis: Optional[int] = -1,
 ) -> Tuple[np.ndarray, np.ndarray]:
-    """Return time-domain rotary components time series (zp,zn) from Cartesian components time series (u,v).
-    Note that zp and zn are both analytic time series which implies that the complex-valued time series
-    u+1j*v is recovered by zp+np.conj(zn). The mean of the original complex signal is split evenly between
-    the two rotary components.
+    """Return rotary signals (wp,wn) from analytic Cartesian signals (ua,va).
+
+    If ua is the analytic signal from real-valued signal u, and va the analytic signal
+    from real-valued signal v, then the positive (counterclockwise) and negative (clockwise)
+    signals are defined by wp = 0.5*(up+1j*vp), wp = 0.5*(up-1j*vp).
```

Comment:
We currently don't have a convention whether to put code snippets in docstrings in code blocks (double backticks) quotes or not. Using code blocks makes them look nicer in HTML docs, but less nice in interpreter. We don't have to resolve this now but let's think about it and be consistent.